### ISSUE TITLE
Populate better solver competition data

### ISSUE THREAD

**Issue by fleupold** [cowprotocol/services#1949]
Publicly available competition data in the colocated setup is very limited at the moment:

![image](https://github.com/cowprotocol/services/assets/1200333/ab3d60a4-2c21-41b1-9138-3e79f8b11e40)

In order to allow insights as to what orders would have matched how and by whom, we should have for every solver:
- score
- orders they intend to execute and their in/out amounts
- uniform clearing price (unless we get rid of it in the future)
- for debugging purpose a readable name would be nice as well

This data should be provided as part of the `solve` response. We still only want to call reveal (for the calldata) on the winning solver.

**Comment by fleupold** [comment]
Probably not blocking launch, but can we show the clearing prices and orders for all solution candidates so that it's somewhat possible to reason about "what if another solver had matched me" ?

**Comment by fhenneke** [comment]
A few test in the EBBO repo depend on being able to compute executed executed sell, buy fee amounts of orders from all the different solutions.
The task might not be blocking in a strict sense. But having this data in the transition from our current mechanism to rank-by-surplus would help assess the impact of that change.

**Comment by fleupold** [comment]
To clarify, do you need the fee amounts as well or would the buy and sell amounts suffice (assuming you also have the clearing prices of the alternative solutions)?

**Comment by fhenneke** [comment]
For most purposes we need the amounts users send us and the amounts users receive, i.e. sell amount (before fees!) and buy amount (after fees!). Having this information for all solutions is most important.
For some purposes, we also need information on fees and clearing prices. Fees can, for example, be reconstructed from sell/buy amounts in combination with clearing prices. And clearing prices can be reconstructed from sell/buy amounts in combination with fee amounts. Which of the two we store is not that important to me. Maybe clearing prices are a bit more fundamental.
In short: sell amounts, buy amounts, and clearing prices are sufficient.
